### PR TITLE
Hide some parser implementations

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: lib/fizzy/parser.hpp [hunk 1]
```diff
@@ -24,40 +24,6 @@ parser_result<Code> parse_expr(const uint8_t* input);
...
-    case 0x7F:
-        return {ValType::i32, pos};
-    case 0x7E:
-        return {ValType::i64, pos};
-    default:
-        throw parser_error{"invalid valtype " + std::to_string(b)};
-    }
-}
-
 inline parser_result<Limits> parse_limits(const uint8_t* pos)
```
Comment (axic) [inline lib/fizzy/parser.hpp]: How about this?

Comment (chfast) [reply lib/fizzy/parser.hpp]: This is used in unit tests directly.

Comment (axic) [reply lib/fizzy/parser.hpp]: Yes, but will the test be changed (in a new PR)?

Comment (chfast) [reply lib/fizzy/parser.hpp]: Can be done, but adding bounds checking to parser seems more important at the moment.

File: lib/fizzy/parser.hpp [hunk 2]
```diff
@@ -19,19 +19,6 @@ parser_result<Code> parse_expr(const uint8_t* input);
...
-template <>
-inline parser_result<uint8_t> parse(const uint8_t* pos)
-{
-    const auto result = *pos;
-    ++pos;
-    return {result, pos};
-}
-
 template <>
 inline parser_result<uint32_t> parse(const uint8_t* pos)
```
Comment (axic) [inline lib/fizzy/parser.hpp]: How about this?

Comment (chfast) [reply lib/fizzy/parser.hpp]: This is also used in `parser_expr.cpp`.

**Pull request merged**